java/ql/src/semmle/code/java/frameworks/android/SharedPreferences.qll

```diff
@@ -0,0 +1,62 @@
+/** Provides classes related to `android.content.SharedPreferences`. */
+
+import java
+
+/** Definitions related to `android.content.SharedPreferences`. */
+module SharedPreferences {
+  /** The interface `android.content.SharedPreferences` */
+  class TypeBase extends Interface {
+    TypeBase() { hasQualifiedName("android.content", "SharedPreferences") }
+  }
+
+  /** The class `androidx.security.crypto.EncryptedSharedPreferences`, which implements `SharedPreferences` with encryption support. */
+  class TypeEncrypted extends Class {
+    TypeEncrypted() { hasQualifiedName("androidx.security.crypto", "EncryptedSharedPreferences") }
+  }
+
+  /** The create method of `androidx.security.crypto.EncryptedSharedPreferences` */
+  class CreateEncryptedMethod extends Method {
+    CreateEncryptedMethod() {
+      getDeclaringType() instanceof TypeEncrypted and
+      hasName("create")
```

Comment:
This is a minor style thing, but we generally prefer to use the `this.` qualifier instead of leaving it implicit. Might as well apply this change consistently throughout this file.
```suggestion
      this.hasName("create")
```

Reply:
I've applied this change throughout this file and the relevant query file.